operators/src/source/gdal_source/mod.rs

```diff
@@ -592,15 +593,19 @@ fn read_grid_from_raster<
     T,
     D: GridSize<ShapeArray = [usize; 2]> + GridSpaceToLinearSpace<IndexArray = [isize; 2]>,
 >(
-    rasterband: &GdalRasterBand,
+    dataset: &gdal::Dataset,
     dataset_grid_box: &GridBoundingBox2D,
     tile_grid: D,
-    no_data_value: Option<T>,
+    dataset_params: &GdalDatasetParameters,
 ) -> Result<GridOrEmpty<D, T>>
 where
-    T: Pixel + GdalType + Default,
+    T: Pixel + GdalType + Default + FromPrimitive,
     D: PartialEq + Clone,
 {
+    let raster_band_index = dataset_params.rasterband_channel;
+
+    let rasterband = dataset.rasterband(raster_band_index as isize)?;
```

Comment:
the rasterband is now opened twice(?) (if there is propertiesMapping). Do you know whether there is some cost in doing so?

Reply:
i dont know. We can change it back once we add the mask band getter to the gdal crate

Reply:
ok. can you add a todo in the code?